src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,67 @@
+import { useState } from 'react';
+
+import { IcArrowsChevronLeftFilled, IcArrowsChevronRightFilled } from '@/style';
+
+import { StyledButton, StyledNav } from './Pagination.style';
+import { PaginationProps } from './Pagination.type';
+
+export const Pagination = ({ totalPage, initialPage = 1, setPage }: PaginationProps) => {
+  if (totalPage <= 1) throw new Error('totalPage는 2 이상의 숫자여야 합니다.');
+  if (initialPage > totalPage) throw new Error('initialPage는 totalPage보다 클 수 없습니다.');
+
+  const [currentPage, setCurrentPage] = useState(initialPage);
+
+  const pages = Array.from({ length: totalPage }).reduce<number[][]>(
+    (section, _, index) => {
+      if (index % 5 === 0) section.push([]);
+      section[section.length - 1].push(index + 1);
+      return section;
+    },
+    [[]]
+  );
+  const lastSection = Math.ceil(totalPage / 5);
+  const currentSection = Math.ceil(currentPage / 5);
+  const pagesPerSection = 5;
+
+  const handleClickPage = (page: number) => {
+    setCurrentPage(page);
+    setPage?.(page);
+  };
+
+  const handleClickArrow = (direction: 'left' | 'right') => {
+    const offset = direction === 'left' ? -pagesPerSection : pagesPerSection;
+    handleClickPage(5 * (currentSection - 1) + 1 + offset);
+  };
```

Comment:
currentPage가 6 이상일 때 left arrow 를 누르면
이전 section의 최대값으로 가는게 아니고, 최소값으로 가네요.
=> 따라서 right arrow 클릭시에는 괜찮은데 left arrow 클릭시에는 어색해요.

https://github.com/user-attachments/assets/701c1d54-6765-4309-b975-59d3eb209e8b


아래 코드로 변경해보세요.

```suggestion
  const handleClickArrow = (direction: 'left' | 'right') => {
    const basePageOffset = 5 * (currentSection - 1);
    const pageMovement = direction === 'left' ? -pagesPerSection : pagesPerSection;
    const pageBoundaryOffset = direction === 'left' ? pagesPerSection : 1;
    handleClickPage(basePageOffset + pageMovement + pageBoundaryOffset);
  };
```

https://github.com/user-attachments/assets/46488cb7-d59a-4073-9ed6-354df2e88b52

Reply:
> 이전 section의 최대값으로 가는게 아니고, 최소값으로 가네요.

어 이거는... 의도한 바가 맞습니다!!!!
저는 화살표 버튼을 누르면 방향에 관계 없이 이전/다음의 첫번째 페이지가 선택된다고 이해했는데,
듣고 보니까 또.. 그런 거 같기도

이거 슬랙에 한번만 확인해보고 변경할게요 ㅎ.ㅎ 코드 제안까지 감사합니다~~~!

Reply:
https://yourssu.slack.com/archives/C07A1528KDE/p1723969036870669
여기서 완료

Reply:
대사 뺏겼다 감사감사합니다